### ISSUE TITLE
lint: paramTypeCombine gives false invalid suggestion for unnamed params

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#256]
Reproducer:
```
gocritic check-package -enable=paramTypeCombine bytes
```

Output:
```
src/bytes/bytes.go:946:1: paramTypeCombine: func(sep []byte) (uint32, uint32) could be replaced with func(sep []byte) (uint32)
```
Function:
```go
func hashStr(sep []byte) (uint32, uint32) {
```
The `(uint32, uint32)` != `(uint32)`.

paramTypeCombine should probably skip this case and do not emit warning.
Note that `unnamedResult` will complain about that:
```
bytes/bytes.go:946:1: unnamedResult: consider to give name to results
```
And if advice is followed, new issue can be found by `paramTypeCombine`.

I'm almost sure `paramTypeCombine` should not suggest giving names to types just to combine them,
there is also `unusedParam` checker that suggests to use `_` for every unused param, so there are quite a bit effort around that already.
